src/agent/onefuzz/src/syncdir.rs

```diff
@@ -101,18 +107,66 @@ impl SyncedDir {
         }
     }
 
+    // Conditionally upload a report, if it would not be a duplicate.
+    pub async fn upload<T: Serialize>(&self, name: &str, data: &T) -> Result<bool> {
+        match self.url.as_file_path() {
+            Some(path) => {
+                let path = path.join(name);
+                if !exists(&path).await? {
+                    let data = serde_json::to_vec(&data)?;
+                    fs::write(path, data).await?;
+                    Ok(true)
+                } else {
+                    Ok(false)
+                }
+            }
+            None => {
+                let url = self.url.blob(name).url();
+                let blob = BlobClient::new();
+                let result = blob
+                    .put(url.clone())
+                    .json(data)
+                    // Conditional PUT, only if-not-exists.
+                    // https://docs.microsoft.com/en-us/rest/api/storageservices/specifying-conditional-headers-for-blob-service-operations
+                    .header("If-None-Match", "*")
+                    .send_retry_default()
+                    .await
+                    .context("Uploading blob")?;
+
+                Ok(result.status() == StatusCode::CREATED)
+            }
+        }
+    }
+
     async fn file_monitor_event(&self, event: Event) -> Result<()> {
         debug!("monitoring {}", self.path.display());
         let mut monitor = DirectoryMonitor::new(self.path.clone());
         monitor.start()?;
 
-        let mut uploader = self.url.as_ref().map(|x| BlobUploader::new(x.url()));
+        if let Some(path) = self.url.as_file_path() {
+            fs::create_dir_all(&path).await?;
+
+            while let Some(item) = monitor.next().await {
+                event!(event.clone(); EventData::Path = item.display().to_string());
+                let file_name = item
+                    .file_name()
+                    .ok_or_else(|| anyhow!("invalid file path"))?;
+                let destination = path.join(file_name);
+                if let Err(err) = fs::copy(&item, &destination).await {
+                    warn!(
```

Comment:
if fs::copy fails, we should fail hard, otherwise we'll swallow errors in practice

Reply:
This was because when monitoring lib fuzzer crashes i saw an issue where the monitor would pick up what seemed to be a temporary file and the file would disappear before the copy was invoked which would cause an error

Reply:
We handled the race condition for syncing inputs from libFuzzer in the remote mode in #204 .

As a compromise, can we attempt the move, if it fails, check to see if the source still exists.  If the source still exists, then bail?

Reply:
#204 is addressing a different problem. Here is what the error looks like 
```
Error: Couldn't upload file.  source:"/mnt/c/work/onefuzz/src/agent/00000000-0000-0000-0000-000000000000/c00fed1e-cced-46d3-9ccd-1f5d645122a5/inputs_dir/.2f654f29901d09cd755b50314cdbabecb25b43d7.5O4S1O" destination:"/mnt/c/work/onefuzz-samples/examples/simple-libfuzzer_linux/seeds/.2f654f29901d09cd755b50314cdbabecb25b43d7.5O4S1O" err:No such file or directory (os error 2)
``` 
I added a check to see if the file still exists as you suggested